src/megaapi_impl.cpp

```diff
@@ -23107,6 +23148,128 @@ void MegaApiImpl::sendPendingRequests()
+
+            tlvRecords.reset(TLVstore::containerToTLVrecords(deviceNameContainerStr, &client->key));
+            if (!tlvRecords || !tlvRecords->find("n")) { e = API_EINCOMPLETE; break; } // is 'n' the right key ??
+
+            const string& devName64Str = tlvRecords->get("n");
+            string deviceName = Base64::btoa(devName64Str);
+            if (deviceName.empty()) { e = API_EINCOMPLETE; break; }
+
+            // get 'device-id'
+            const string& deviceId = client->getDeviceid();
+            if (deviceId.empty()) { e = API_EINCOMPLETE; break; }
+
+            // loop remote children of "My Backups" folder, to find `DEVICE_NAME`-with-'device-id' node
+            const MegaNodeList* nodeList = myBackupsNode->getChildren();
+            MegaNode* deviceNameNode = nullptr;
+            for (int i = 0; i < nodeList->size(); ++i)
+            {
+                MegaNode* child = nodeList->get(i);
+
+                // get folder name
+                const char* name = child->getName();
+                bool gotName = deviceName == name;
+
+                // get 'device-id' tag
+                const char* id = child->getCustomAttr("device-id");
+                bool gotId = deviceId == id;
+
+                // determine whether we found our node or not
+                if (!gotName && !gotId) { continue; } // ignore this child folder
+
+                if (gotName && gotId) // got it, use this one to add another backup to it
+                {
+                    deviceNameNode = child;
+                    break;
+                }
+
+                // there's an error if only one of the folder's {name, id} is correct and the other is wrong.
+                e = API_EINCOMPLETE;
+                break;
+            }
+
+            if (e != API_OK) { break; }
+
+            if (!deviceNameNode)
+            {
+                // create "My Backups"/`DEVICE_NAME`/
+                // HOW ? Like this?
+                auto nextSyncTag = client->nextSyncTag();
+                auto sync = make_unique<MegaSyncPrivate>(nullptr, deviceName.c_str(), h, nextSyncTag);
+                sync->setListener(request->getSyncListener());
+
+                // NOW WHAT ? Something like this?
+                std::unique_ptr<char[]> remotePath{ getNodePathByNodeHandle(h) };
+                SyncConfig syncConfig{ nextSyncTag, nullptr, deviceName.c_str(), h, remotePath.get(),
+                                       0, {}, true, SyncConfig::TYPE_BACKUP };
+
+                SyncError syncError = NO_SYNC_ERROR;
+                e = client->addsync(syncConfig, DEBRISFOLDER, nullptr, syncError, true, sync.get());
+
+                fireOnSyncAdded(sync.get(), MegaSync::NEW);
+                fireOnRequestFinish(request, make_unique<MegaErrorPrivate>(e));
+
+                // Update deviceNameNode somehow?
+                // HOW ?
+
+                // tag the new node with `dev:<device-id>` in `attrstring`
+                // HOW ?
```

Comment:
It would be something like:
``` cpp
node->attrs.map['d'] = device_id;
```
It would set the device id in the map of attributes and those attributes will be serialized and encrypted before sending to API.

Reply:
Uhm, it's still not clear what would make the difference between `dev:<device-id>` and `bak:<device-id>`.

Reply:
Where that `bak:<device-id>` comes from?

Reply:
Last comment in Jira SDK-1243

Reply:
Oh, I see. It's just a suggestion. It should be the same thing, we are tagging the node with a node's attribute that stores the device-id. Simply use the same tag, both for the `DEVICE_NAME` folder and for the backup folder. Both folder nodes should add the `device-id` in a node's attribute. The tag is irrelevant.